tests/data/dataset_readers/wikitables/wikitables_dataset_reader_test.py

```diff
@@ -11,17 +12,244 @@ def test_reader_reads(self):
+                "<d,<<d,d>,d>> -> [<d,<d,<<d,d>,d>>>, d]",
+                "<d,<d,<#1,<<d,#1>,#1>>>> -> argmax",
+                "<d,<d,<#1,<<d,#1>,#1>>>> -> argmin",
+                "<d,<d,<<d,d>,d>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<d,d>> -> -",
+                "<d,<e,<<d,e>,e>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<p,<<d,p>,p>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<r,<<d,r>,r>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+
+                # Operations that manipulate numbers.
+                # TODO(mattg): question for Pradeep: why aren't these two-argument functions?
+                "<d,d> -> <",
+                "<d,d> -> <=",
+                "<d,d> -> >",
+                "<d,d> -> >=",
+                "<d,d> -> ['lambda x', d]",
+                "<d,d> -> [<#1,<#1,#1>>, d]",
+                "<d,d> -> [<<#1,#2>,<#2,#1>>, <d,d>]",
+                "<d,d> -> [<d,<d,d>>, d]",
+                # These are single-argument functions because the "d" type here actually represents
+                # a set, and we're performing some aggregation over the set and returning a single
+                # number (actually a set with a single number in it).
+                "<d,d> -> avg",
+                "<d,d> -> max",
+                "<d,d> -> min",
+                "<d,d> -> sum",
+                "<d,e> -> ['lambda x', e]",
+                "<d,e> -> [<<#1,#2>,<#2,#1>>, <e,d>]",
+                # TODO(mattg): isn't this backwards?  Aren't these functions that take cells and
+                # return numbers?
+                "<d,e> -> fb:cell.cell.date",
+                "<d,e> -> fb:cell.cell.num2",
+                "<d,e> -> fb:cell.cell.number",
+                "<d,p> -> ['lambda x', p]",
+                "<d,p> -> [<<#1,#2>,<#2,#1>>, <p,d>]",
+                "<d,r> -> ['lambda x', r]",
+                "<d,r> -> [<<#1,#2>,<#2,#1>>, <r,d>]",
+                "<d,r> -> fb:row.row.index",
+
+                # Now we get to the CELL_TYPE, which is represented by "e".
+                "<e,<<d,e>,e>> -> [<d,<e,<<d,e>,e>>>, d]",
+                # TODO(mattg): why is this so complicated?
+                "<e,<e,<e,d>>> -> date",
+                "<e,<e,d>> -> [<e,<e,<e,d>>>, e]",
+                "<e,d> -> ['lambda x', d]",
+                "<e,d> -> [<<#1,#2>,<#2,#1>>, <d,e>]",
+                "<e,d> -> [<e,<e,d>>, e]",
+                "<e,d> -> number",
+                "<e,e> -> ['lambda x', e]",
+                "<e,e> -> [<#1,<#1,#1>>, e]",
+                "<e,e> -> [<<#1,#2>,<#2,#1>>, <e,e>]",
+                "<e,p> -> ['lambda x', p]",
+                "<e,p> -> [<<#1,#2>,<#2,#1>>, <p,e>]",
+                "<e,r> -> ['lambda x', r]",
+                "<e,r> -> [<<#1,#2>,<#2,#1>>, <r,e>]",
+
+                # These are instance-specific production rules.  These are the columns in the
+                # table.
+                # TODO(mattg): this also seems backwards.
```

Comment:
Columns are relations from cells to rows.

Reply:
Yeah, backwards.  But at least now I know =).